spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/pubsub/GcpPubSubAutoConfigurationTests.java

```diff
@@ -163,13 +168,57 @@ public void executorThreads_globalAndSelectiveConfigurationSet_selectiveTakesPre
 			GcpPubSubProperties gcpPubSubProperties = ctx
 					.getBean(GcpPubSubProperties.class);
 			GcpProjectIdProvider projectIdProvider = ctx.getBean(GcpProjectIdProvider.class);
-			assertThat(gcpPubSubProperties.getSubscriber("subscription-name", projectIdProvider.getProjectId())
-					.getExecutorThreads()).isEqualTo(3);
+			assertThat(
+					gcpPubSubProperties.computeSubscriberExecutorThreads("subscription-name", projectIdProvider.getProjectId()))
+							.isEqualTo(3);
 			assertThat(gcpPubSubProperties.getSubscription())
 					.containsKey("projects/fake project/subscriptions/subscription-name");
 			assertThat(
-					gcpPubSubProperties.getSubscriber("other", projectIdProvider.getProjectId()).getExecutorThreads())
+					gcpPubSubProperties.computeSubscriberExecutorThreads("other", projectIdProvider.getProjectId()))
+							.isEqualTo(5);
+		});
+	}
+
+	@Test
+	public void executorThreads_globalAndDifferentSelectiveConfigurationSet_pickGlobal() {
```

Comment:
```suggestion
	public void executorThreads_inheritsGlobalConfig() {
```

nit: suggestion to rename test (maybe is more concise)

Reply:
Nevermind, I think the way you had better matches the existing naming conventions.